src/app/home/avatar-generator/selector/atribute-selector.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, OnInit, Input, Inject } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import {AvatarDecoratorItem, AvatarOptions} from '../Avatartypes';
+import {buildAvatar} from '../AvatarUtils';
+import {getPreview} from '../AvatarSelectorPath';
+
+@Component({
+  // tslint:disable-next-line: component-selector
+  selector: 'atribute-selector',
+  templateUrl: './atribute-selector.html',
+  styleUrls: ['./atribute-selector.scss'],
+})
+export class AtributeSelector implements OnInit {
+  // @Output() newValue = new EventEmitter<boolean>();
+  @Input() avatar: AvatarOptions;
+  @Input() decorator: AvatarDecoratorItem;
+  @Input() dialogId: string;
+  modal: any;
+  color: string;
+  constructor(@Inject(DOCUMENT) document) { }
+
+  openModal(): void {
+    const modal: any = document.getElementById(this.dialogId);
+    modal.showModal();
+  }
+
+  closeModal(): void {
+    const modal: any = document.getElementById(this.dialogId);
```

Comment:
Please use the reference of `DOCUMENT` injected to this class instead of the global `document` object:
`this.document`